packages/block-library/src/cover/overlay-color-settings.native.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getColorObjectByColorValue,
+	getColorObjectByAttributeValues,
+	getGradientValueBySlug,
+	getGradientSlugByValue,
+	__experimentalPanelColorGradientSettings as PanelColorGradientSettings,
+} from '@wordpress/block-editor';
+
+function OverlayColorSettings( { attributes, setAttributes } ) {
+	const { colors, gradients } = useSelect( ( select ) => {
+		return select( 'core/block-editor' ).getSettings();
+	}, [] );
+
+	const {
+		overlayColor,
+		customOverlayColor,
+		gradient,
+		customGradient,
+	} = attributes;
+
+	const gradientValue =
+		customGradient || getGradientValueBySlug( gradients, gradient );
+
+	const colorValue = getColorObjectByAttributeValues(
+		colors,
+		overlayColor,
+		customOverlayColor
+	).color;
+
+	const setOverlayAttribute = ( attributeName, value ) => {
+		setAttributes( {
+			...attributes,
```

Comment:
Just wondering why do we need to spread the current attributes 🤔I don't think we need that, what do you think?

Reply:
You're right, that's not necessary :smile:, nice catch! I've removed it.